zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/transfer/SnapshotTransferServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.snapshots.transfer;
+
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import io.camunda.zeebe.scheduler.future.CompletableActorFuture;
+import io.camunda.zeebe.snapshots.ConstructableSnapshotStore;
+import io.camunda.zeebe.snapshots.SnapshotChunk;
+import io.camunda.zeebe.snapshots.SnapshotChunkReader;
+import io.camunda.zeebe.snapshots.impl.FileBasedSnapshotChunkReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class SnapshotTransferServiceImpl implements SnapshotTransferService {
+
+  private final ConstructableSnapshotStore snapshotStore;
+  private final Map<UUID, PendingTransfer> pendingTransfers = new HashMap<>();
+  private final int partitionId;
+
+  public SnapshotTransferServiceImpl(
+      final ConstructableSnapshotStore snapshotStore, final int partitionId) {
+    this.snapshotStore = snapshotStore;
+    this.partitionId = partitionId;
+  }
+
+  @Override
+  public ActorFuture<SnapshotChunk> getLatestSnapshot(final int partition, final UUID transferId) {
```

Comment:
:question: There's nothing async here, why the signature?

Reply:
The interface uses `ActorFuture` so that it can be implemented over local storage or over the network. 
I think the mistake here is that the calls to the `SnapshotChunkReader` should be run in an IO actor thread pool.
I will make sure they are not run in the caller thread.